core/src/main/java/org/apache/iceberg/BaseReplacePartitions.java

```diff
@@ -87,24 +80,26 @@ public BaseReplacePartitions toBranch(String branch) {
 
   @Override
   public void validate(TableMetadata currentMetadata, Snapshot parent) {
+    super.validate(currentMetadata, parent);
+
     if (validateConflictingData) {
       if (dataSpec().isUnpartitioned()) {
         validateAddedDataFiles(
-            currentMetadata, startingSnapshotId, Expressions.alwaysTrue(), parent);
+            currentMetadata, startingSnapshotId(), Expressions.alwaysTrue(), parent);
```

Comment:
This is a good example of what I mean in the comment above. There are quite a few changes here (and in other classes) to refactor so that `startingSnapshotId` can be reused.